service/service.go

```diff
@@ -62,7 +61,8 @@ type Service struct {
 
 	// openRound is the round which is currently open for accepting challenges registration from miners.
 	// At any given time there is one single open round.
-	openRound *round
+	openRound      *round
+	openRoundMutex sync.Mutex
```

Comment:
How about using an atomic pointer instead?:
```suggestion
        //SAFETY: the contents behind the pointer are read-only
	openRound atomic.Pointer[round]
```

---
Edit: After looking further in the code it seems that this mutex doesn't just guard `openRound` variable only (it is locked for some functions invocations). Please consider documenting what this mutex guards.

Reply:
Yes, atomic.Pointer wouldn't work without some bigger changes. I changed `openRoundMutex` to be a `sync.RWMutex` so multiple routines can read the ID of the current round.